src/main/java/seedu/address/model/task/TaskList.java

```diff
@@ -0,0 +1,132 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.task.execeptions.TaskNotFoundException;
+
+/**
+ * Represents a list of delivery tasks. The `TaskList` class supports a minimal set
+ * of list operations. List sort and filter operations are provided.
+ */
+public class TaskList {
+
+    private static int lastTaskId;
+    private final ObservableList<Task> tasks = FXCollections.observableArrayList();
+    private final ObservableList<Task> tasksUnmodifiable = FXCollections.unmodifiableObservableList(tasks);
+
+    public TaskList() {
+        lastTaskId = 0;
```

Comment:
Static variable's values should not be initiated in class constructor.. You will never know who will call another `TaskList()` somewhere and just throw it away.. Init in the attribute field.